### ISSUE TITLE
API GET AUTH TOKEN ERROR: Cannot read properties of undefined (reading 'authorizationUrl')

### ISSUE THREAD

**Issue by MKniet** [iobroker-community-adapters/ioBroker.wolf-smartset#304]
1. Install Wolf-smartset
2. Add User and PW
3. Touch get DeviceList
4. --> Error messages

 wolf-smartset.0 | 2023-11-25 16:31:55.118 | info | getDeviceList

 wolf-smartset.0 | 2023-11-25 16:31:55.120 | warn | API GET AUTH TOKEN ERROR: Cannot read properties of undefined (reading'authorizationUrl')

wolf-smartset.0 | 2023-11-25 16:31:55.123 | warn | TypeError: Cannot read properties of undefined (reading 'authorizationUrl') at WolfSmartSet._getAuthToken (/opt/iobroker/node_modules/iobroker.wolf-smartset/lib/wss.js:167:38) at WolfSmartSet.adminGetDevicelist (/opt/iobroker/node_modules/iobroker.wolf-smartset/lib/wss.js:134:32) at WolfSmartset.onMessage (/opt/iobroker/node_modules/iobroker.wolf-smartset/main.js:430:34) at WolfSmartset.emit (node:events:514:28) at WolfSmartset.emit (node:domain:489:12) at change (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11134:34) at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/db-states-redis/src/lib/states/statesInRedisClient.ts:379:37) at processImmediate (node:internal/timers:476:21)

Final: i not know hot to fix this. Please help me =) 
Thank you in advence

**Comment by Klaus-Lucas** [comment]
@MKniet 
Do you read #286?

**Comment by MKniet** [comment]
hi i try also the Version 1.0.0 i have the problem he not accept my user and PW 

CPU: 7,93 %
RAM: 76,5 %
Betriebszeit: 1d0h
Verfügbar: 5.0.16
Installiert: 5.0.16
Ereignisse: ⇥22 / ↦15
Plattform: linux
Betriebssystem: linux
Architektur: arm64
CPUs: 4
Geschwindigkeit: 1800 MHz
Modell: Cortex-A72
RAM: 7.6 GB
System-Betriebszeit: 85 T. 18:41:15
Node.js: v18.17.1
time: 1700990748930
timeOffset: -60
NPM: 9.6.7
Adapter-Anzahl: 515
Datenträgergröße: 117.0 GB
Freier Festplattenspeicher: 112.6 GB
Aktive Instanzen: 23
Pfad: /opt/iobroker/
Betriebszeit: 23:46:24
aktiv: true
_nodeCurrent: 18.17.1
_nodeNewest: 18.18.2
_nodeNewestNext: 18.18.2
_npmCurrent: 9.6.7
_npmNewest: 9.6.7
_npmNewestNext: 9.8.1



![Screenshot 2023-11-26 104033](https://github.com/iobroker-community-adapters/ioBroker.wolf-smartset/assets/152018900/3ab234b8-9e64-4c5d-90db-884081f5b07f)

**Comment by Klaus-Lucas** [comment]
@MKniet 
I had the same issue. Steps that fixed it:
- Remove adapter 1.1.1
- Install 1.1.1 again and skip the configuration page
- Downgrade to 1.0.0 and enter the same credentials you use on the Wolf Smartset page and select your device
- Update to 1.1.1 again

Logins via API to the Wolf Smarset portal has still negative responses but they will happen only in an intervall, depending on your configured query intervall. Please see #302, where I decribed this.

**Comment by MKniet** [comment]
Thank you !! Now it is Working!!

**Comment by bstmedia** [comment]
I got the same issue.  Downgrade to 1.0.0 did not help neither.

API GET AUTH TOKEN ERROR: Cannot read properties of undefined (reading 'authorizationUrl')

TypeError: Cannot read properties of undefined (reading 'authorizationUrl') at WolfSmartSet._getAuthToken (/opt/iobroker/node_modules/iobroker.wolf-smartset/lib/wss.js:167:38) at WolfSmartSet.adminGetDevicelist (/opt/iobroker/node_modules/iobroker.wolf-smartset/lib/wss.js:134:32) at WolfSmartset.onMessage (/opt/iobroker/node_modules/iobroker.wolf-smartset/main.js:430:34) at WolfSmartset.emit (node:events:517:28) at WolfSmartset.emit (node:domain:489:12) at change (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11114:34) at Immediate._onImmediate (/opt/iobroker/node_modules/@iobroker/db-states-redis/src/lib/states/statesInRedisClient.ts:379:37) at processImmediate (node:internal/timers:476:21)

Is there anything else i could try ?

Thanks
Stefan

**Comment by Klaus-Lucas** [comment]
@bstmedia Schick mal deine Kontaktdaten zu DiddisDwl@web.de.
Ggf. kann man per TeamViewer das Problem lösen.

**Comment by bstmedia** [comment]
I did work with downgrade to 1.0.0 and upgrade to 1.1.1 . Not sure why it failed yesterday.

**Comment by Saschaisi** [comment]
Hi, since a few Days I can not get a working connection to the smart Set Protal. I Tried the steps from above, but i can not get the device. The Trick with Downgrade does not work, I always get Error 400 in the log with 1.0.0. Attached the Log.

Adapter version: 1.1.1
Admin: v6.13.16
JS-Controller version: 5.0.19
Node version: v18.17.1
NPM: 9.6.7
Operating system: linux Raspberry Pi OS

![grafik](https://github.com/iobroker-community-adapters/ioBroker.wolf-smartset/assets/54538020/a744f7ec-cf42-45dd-99b2-a620cf32e7f6)

**Comment by flingo64** [comment]
Same here: trying to connect the WolfSmartSet since two or three days without success. Did try the downgrade/upgrade work-around, also no success.
SystemInfo (Docker Container on Synology NAS):
```
    Plattform: docker (official image - v9.1.2)
    Betriebssystem: linux
    Architektur: x64
    Node.js: v18.20.1
    NPM: 10.5.1

```

**Comment by flingo64** [comment]
I did some debugging of the adapter code and added - besides some debug output - basically one line of code which fixed it for me:

In file lib/wss.js, in function _getAuthToken immediately before the try/catch block:
`		this.openIdInit()
`
Debugging showed that openIdInit() was never called before doing _getAuthToken, so `this.openIdClient` was not yet set when calling getDeviceList, which is what the TypeError in the log complains about.

I published a [patch version](https://github.com/flingo64/ioBroker.wolf-smartset/archive/refs/tags/v1.1.3.tar.gz) with the adapter named  'iobroker.wolf-ism7' so one can install both adapters side-by-side and check the difference.

Feel free to check out the patch. If this fixes the issue for others also I will do a merge request with the one line changed.

**Comment by mcm1957** [comment]
Thanks for your effort. 
Please provide a PR if you think the change will fix the issue.

**Comment by Apollon77** [comment]
If it is just this then the fix is a one liner. Thank you for trying it ... provide PR and we will update !

**Comment by flingo64** [comment]
PR will  follow on Monday, based on latest release .1.2.0.

**Comment by mcm1957** [comment]
fixed at 1.2.1

**Comment by flingo64** [comment]
Thanks!!
